### ISSUE TITLE
[heft] No linting during `heft start`

### ISSUE THREAD

**Issue by Faithfinder** [microsoft/rushstack#3381]
Follow up from [Zulip](https://rushstack.zulipchat.com/#narrow/stream/262522-heft/topic/heft.20start.20-.20no.20linting.3F) question:

Currently `heft start` doesn't show linting errors.
Personally, I would like it to do that, otherwise there's a high chance the developers will commit them and only find out when CI runs `build`. Doesn't have to block the dev server, but showing errors in console would be very helpful.

**Comment by dmichon-msft** [comment]
Currently by design but will be revisited in the current redesign that Heft is undergoing.

**Comment by dmichon-msft** [comment]
There's a complication here in that Heft attempted to do incremental linting, however the list of changed files that need to be emitted according to TypeScript is apparently not quite the same as the list of files that need to be re-linted. The main reason linting is disabled in watch mode is because it is expensive; however with linting potentially running in parallel to bundling this may be less of a concern.

**Comment by Faithfinder** [comment]
I'm very happy with the experience of using Vite with this plugin: https://www.npmjs.com/package/vite-plugin-checker

TLDR - you get output/app runs regardless of linting and typechecking, and those are running in parallel in a worker thread, showing you the results as they become ready.
